File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {

        int c = 0;
        int n = nums.size();
        if (nums[0] < nums[n - 1]) {
            c++;
        }
        for (int i = 0; i < n - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                c++;
            }
            if (c > 1) {
                return false;
            }
        }
        return true;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        cnt = 0
        s = (len(nums) - 1)
        if nums[s] > nums[0] : cnt += 1
        for i in range(1, s + 1):
            if nums[i-1] > nums[i] : cnt += 1
            if cnt > 1 : break
        return cnt <= 1
